src/test/java/org/projectnessie/tools/catalog/migration/CLITest.java

```diff
@@ -0,0 +1,525 @@
+import static org.projectnessie.tools.catalog.migration.CatalogMigrationCLI.FAILED_IDENTIFIERS_FILE;
+import static org.projectnessie.tools.catalog.migration.CatalogMigrationCLI.FAILED_TO_DELETE_AT_SOURCE_FILE;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class CLITest {
+
+  private static @TempDir File warehouse1;
+  private static String warehousePath1;
+
+  private static @TempDir File warehouse2;
+  private static String warehousePath2;
+
+  private static Catalog catalog1;
+
+  private static Catalog catalog2;
+
+  private static final Schema schema =
+      new Schema(Types.StructType.of(required(1, "id", Types.LongType.get())).fields());
+
+  @BeforeAll
+  protected static void setup() {
+    warehousePath1 = String.format("file://%s", warehouse1.getAbsolutePath());
+    warehousePath2 = String.format("file://%s", warehouse2.getAbsolutePath());
+
+    catalog1 = createCatalog(warehousePath1, "catalog1");
+    ((HadoopCatalog) catalog1).createNamespace(Namespace.of("foo"), Collections.emptyMap());
+    ((HadoopCatalog) catalog1).createNamespace(Namespace.of("bar"), Collections.emptyMap());
+
+    catalog2 = createCatalog(warehousePath2, "catalog2");
+    ((HadoopCatalog) catalog2).createNamespace(Namespace.of("foo"), Collections.emptyMap());
+    ((HadoopCatalog) catalog2).createNamespace(Namespace.of("bar"), Collections.emptyMap());
+  }
+
+  @BeforeEach
+  protected void beforeEach() {
+    // two tables in 'foo' namespace
+    catalog1.createTable(TableIdentifier.of(Namespace.of("foo"), "tbl-1"), schema);
+    catalog1.createTable(TableIdentifier.of(Namespace.of("foo"), "tbl-2"), schema);
+    // two tables in 'bar' namespace
+    catalog1.createTable(TableIdentifier.of(Namespace.of("bar"), "tbl-3"), schema);
+    catalog1.createTable(TableIdentifier.of(Namespace.of("bar"), "tbl-4"), schema);
+
+    // to handle the user prompt
+    respondAsContinue();
+  }
+
+  private static void respondAsContinue() {
+    String input = "yes\n";
+    ByteArrayInputStream in = new ByteArrayInputStream(input.getBytes());
+    System.setIn(in);
```

Comment:
Definite no go to manipulate the systems standard input or output or error streams as it can easily break test infrastructure.